### PR TITLE
Robo-DTT every half hour from 7 AM PST to 7 PM PST.

### PR DESCRIPTION
**Pull request by ashercodeorg**

### CODE REVIEW

**Comment by ashercodeorg** [comment]
Presumably, this PR should necessitate some changes in the DOTD script. That said, it isn't entirely clear to me how the DOTD script should be structured in the context of this change. Suggestions welcome, I'll add another commit to this PR adjusting the DOTD script appropriately.

**Comment by islemaster** [comment]
I suppose the first change is that the DTT steps basically go away from our DOTD script, replaced by a 'deploy the last green DTT' or 'wait for the next green DTT' step?

One slight complication here is that we currently prompt the DOTD to do a levelbuilder content scoop and then DTT - we explicitly include that content in a DTT for the day, which seems like it might not happen if the timing was off on a given day under this system.  That's probably fine (the content would go out the next day, or if urgent we'd do a second release) but I think it should be called out.

It's going to be trickier for the script to tell DOTD when to chase down DTT failures and such manually.  I know the ideal is that we don't _need_ to do this because the validation gets automated, but maybe for now we just have the script print a note / wait in a few places for some manual DTT validation to occur?

**Comment by ashercodeorg** [comment]
SG, I'll update the DOTD script to remind the user to perform necessary reruns, marking the DTT green if appropriate. In addition to the levelbuilder content scoop, there is also the issue of the staging content scoop. These can be done "anytime", with a later DTT grabbing those commits. As long as this is done at the "normal" time, I'd expect that content to be DTPed like it historically has.

**Comment by ashercodeorg** [comment]
My thinking with 886808b is that the two initial DTTs will encompass the "robo-DTT" and the "AM DTT", then content grabbing, then the "PM DTT". Of course, marking more robo-DTTs as green is icing on the cake.

**Comment by ashercodeorg** [comment]
It seems worth confirming that we pay a flat rate for our usage of `test` (circle, etc.), that increasing the number of DTTs won't touch the bottom line. Right?

File: bin/dotd [hunk 1]
```diff
@@ -337,7 +306,7 @@ def main
     end
   end
 
-  do_dtt_subroutine
+  dtt_subroutine
```
Comment (islemaster) [inline bin/dotd]: You've replaced `do_dtt_subroutine` with `dtt_subroutine`, but the method you've introduced is called `dtt_routine`.

Maybe it's time to unit-test this script :grin:

File: bin/dotd [hunk 2]
```diff
@@ -205,6 +205,25 @@ def content_push(server_name, environment_name)
   end
 end
 
+# Asks if the robo-DTT is green, if the DTT message should be reset, and if Honeybadger issues
+# should be shown.
+def dtt_routine
```
Comment (islemaster) [inline bin/dotd]: For what it's worth, while `routine` and `subroutine` are both correct terms for a 'callable unit', I think `subroutine` is more common and less ambiguous.  At least, it got the title position [on Wikipedia](https://en.wikipedia.org/wiki/Subroutine).  Alternatively, I think `method` is the correct term in ruby - and all of the above are probably considered rendant as part of a method name, esp. in the Ruby idiom.

File: bin/dotd (hunk 1 above)
Comment (ashercodeorg) [reply bin/dotd]: It was time a long time ago. :)

Though to be fair, the length of the script keeps decreasing, as more and more gets pulled out (e.g., `InfraTestTopic`, `GitHub`, etc.) and tested. That said, I'll follow this PR with some unit tests.

File: bin/dotd (hunk 2 above)
Comment (ashercodeorg) [reply bin/dotd]: Renamed to `finish_robo_dtt`, which I think is more descriptive.

**Comment by ashercodeorg** [comment]
PTAL.

File: bin/dotd [hunk 3]
```diff
@@ -267,56 +276,16 @@ def get_honeybadger_issues
...
 
-  should_i 'mark robo-DTT as green' do
-    test_sha = GitHub.sha 'test'
-    InfraTestTopic.set_green_commit test_sha
-  end
-
-  should_i 'reset DTT message to yes' do
-    DevelopersTopic.set_dtt 'yes'
+  2.times do |_i|
+    finish_robo_dtt
```
Comment (islemaster) [inline bin/dotd]: Why the parameter here?  Can't we just `2.times { finish_robo_dtt }`?

Comment (ashercodeorg) [reply bin/dotd]: Because I often author horrible code. Fixing.

**Pull request merged**